extensions/amp-access/0.1/jwt.js

```diff
@@ -59,6 +70,51 @@ export class JwtHelper {
   }
 
   /**
+   * Whether the signature-verification supported on this platform.
+   * @return {boolean}
+   */
+  isVerificationSupported() {
+    return !!this.subtle_;
+  }
+
+  /**
+   * Decodes HWT token and verifies its signature.
+   * @param {string} encodedToken
+   * @param {string} keyUrl
+   * @return {!Promise<!JSONObject>}
+   */
+  decodeAndVerify(encodedToken, keyUrl) {
+    if (!this.subtle_) {
+      throw new Error('Crypto is not supported on this platform');
+    }
+    let decoded;
+    try {
```

Comment:
Wanted to avoid deep nesting of `then` and thus I needed `decoded` on the top level. Which one would look better to you? In the end, I need to have a key and a decoded struct at a point where I do verify.